testing/wait_for_backend_and_run_e2etests.sh

```diff
@@ -6,4 +6,5 @@
 set -ex
 timeout 240 sh -c "while ! curl http://proxy:80/api/actuator/health/ping 2>/dev/null | grep -q UP; do echo 'backend server not yet there - retrying in 1s'; sleep 1; done; echo 'backend server responded'"
 timeout 240 sh -c "while ! curl http://proxy:80/api/actuator/health/skyminderDummyServer 2>/dev/null | grep -q UP; do echo 'skyminder dummy server not yet there - retrying in 1s'; sleep 1; done; echo 'skyminder dummy server responded'"
```

Comment:
`http://proxy:80/api/actuator/health/skyminderDummyServer`

Hä? Die URL sieht irgendwie komisch aus. Also bei `/api/actuator/health` komme ich beim Backend raus, aber bei `/api/actuator/health/skyminderDummyServer` komme ich beim Skyminderserver raus? Das ist komisch. 

Haben wir im Backend eine Funktion, die die Existenz des skyminderServer überprüft?

Reply:
aber ich denke mal http://proxy:80/api/actuator/health sollte ausreichend sein, da es UP ist wenn alle UP sind

Reply:
Stimmt. Aber was ich so wie es aktuell ist nett finde ist, dass du im Log auch siehst welcher Service genau nicht UP war, wenn es hier einen Error gibt.
Diese Information würde ja verloren gehen, wenn du es mit einem einzigen Check machst, oder? Was meinst du?

Reply:
Ich hatte an sowas gedacht:
```
#!/bin/sh
# This script continuously validates, whether the backend-container is running.
# Then it continuously validates, whether the skyminder-dummy-server-container is running.
# Only then it executes the end-to-end tests.

set -ex

is_infrastructure_up () {
  if ! curl http://proxy:80/api/actuator/health/ping 2>/dev/null | grep -q UP
  then
    echo "Backend not yet there"
    return 1
  fi
  if ! curl http://proxy:80/api/actuator/health/skyminderDummyServer 2>/dev/null | grep -q UP
  then
    echo "Skyminder Dummyserver not yet there"
    return 1
  fi
  if ! curl http://proxy:80/api/actuator/health/edcDummyServer 2>/dev/null | grep -q UP
  then
    echo "EDC Dummyserver not yet there"
    return 1
  fi
  return 0
}

timeout 240 sh -c "while ! is_infrastructure_up; do echo 'infrastructure not yet completely there - retrying in 1s'; sleep 1; done; echo 'infrastructure up!'"
./gradlew :dataland-e2etests:test :dataland-frontend:npm_run_testpipeline --no-daemon
```

Reply:
>angefangen zu evaluieren ob es klappt und es umzusetzen<

Reply:
fixed gemeinsam mit Siegfried im `wait_for_backend_and_run_e2etests.sh`

Bitte nochmal reviewen, da etwas abweichend von deinem Vorschlag umgesetzt.

Reply:
Paar Eckpunkte:

- Bash Script statt wie bisher .sh (damit man die Funktion `is_infrastructure_up()` auch in die Sub-Shell exporten kann)
- for-loop statt aufeinanderfolgende if-Blöcke
- läuft über ein Array aus service-Namen (als Key) und service-URLs (als Value)

Reply:
ja - super. danke!